docs/docs/api/Dispatcher.md

```diff
@@ -1097,3 +1097,200 @@ or
   }
 }
 ```
+
+# Response Error Interceptor
+
+## Introduction
+
+The Response Error Interceptor is designed to handle HTTP response errors efficiently. It intercepts responses and throws detailed errors for responses with status codes indicating failure (4xx, 5xx). This interceptor enhances error handling by providing structured error information, including response headers, data, and status codes.
+
+## ResponseError Class
+
+The `ResponseError` class extends the `UndiciError` class and encapsulates detailed error information. It captures the response status code, headers, and data, providing a structured way to handle errors.
+
+### Definition
+
+```js
+class ResponseError extends UndiciError {
+  constructor (message, code, { headers, data }) {
+    super(message);
+    this.name = 'ResponseError';
+    this.message = message || 'Response error';
+    this.code = 'UND_ERR_RESPONSE';
+    this.statusCode = code;
+    this.data = data;
+    this.headers = headers;
+  }
+}
+```
+
+## Interceptor Handler
+
+The interceptor's handler class extends `DecoratorHandler` and overrides methods to capture response details and handle errors based on the response status code.
+
+### Methods
+
+- **onConnect**: Initializes response properties.
+- **onHeaders**: Captures headers and status code. Decodes body if content type is `application/json` or `text/plain`.
+- **onData**: Appends chunks to the body if status code indicates an error.
+- **onComplete**: Finalizes error handling, constructs a `ResponseError`, and invokes the `onError` method.
+- **onError**: Propagates errors to the handler.
+
+### Definition
+
+```js
+class Handler extends DecoratorHandler {
```

Comment:
This can be added under the `interceptor` section to explain what can be the usage of it.

Reply:
applied thank you very much for the feedback